### ISSUE TITLE
Email input on cc payment form presents no error for invalid input

### ISSUE THREAD

**Issue by youriwims** [mozilla/donate-wagtail#366]
### Issue: Email input on cc payment form presents no error as long as '@' is included 
![form-errors(1)](https://user-images.githubusercontent.com/36484608/66056948-e91b7a00-e505-11e9-85b2-bd77bd0fd5ac.gif)

**How to fix:**

- Please add clear error logic that only accepts valid email forms and flags anything else as an error
- Add error state styles as well (_although I assume this will kick in once logic is fixed_)

====

More Info:

Comments: n/a
Resources: Related to #343
cc @philwakefield

**Comment by solarissmoke** [comment]
The email address you have used to test (`test@q`) is considered valid by the [algorithm defined in the HTML5 specification](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/email#Basic_validation), so browsers will accept it. Form validation in Django will fail however, so the user will be given an error when they submit.

We had discussed more robust email validation (including things like checking for common typos - e.g., gmal.com) with Alan and Will, and IIRC agreed to punt it for the time being.

Passing this to Alan to confirm what you'd like to have done on this before launch.

**Comment by alanmoo** [comment]
Yeah, we're gonna leave this as-is for now.
